src/inertial_sense_ros.cpp

```diff
@@ -278,6 +282,68 @@ void InertialSenseROS::start_rtk_server(const std::string& RTK_server_IP, const
 
 }
 
+void InertialSenseROS::start_rtk_connectivity_watchdog_timer()
+{
+  bool rtk_connectivity_watchdog_enabled;
+  // default is false for legacy compatibility
+  nh_private_.param<bool>("RTK_connectivity_watchdog_enabled", rtk_connectivity_watchdog_enabled, false);
+  if (!rtk_connectivity_watchdog_enabled)
+  {
+    return;
+  }
+
+  if (!rtk_connectivity_watchdog_timer_.isValid())
+  {
+    float rtk_connectivity_watchdog_timer_frequency;
+    nh_private_.param<float>("RTK_connectivity_watchdog_timer_frequency", rtk_connectivity_watchdog_timer_frequency, 1);
+    rtk_connectivity_watchdog_timer_ = nh_.createTimer(ros::Duration(rtk_connectivity_watchdog_timer_frequency), InertialSenseROS::rtk_connectivity_watchdog_timer_callback , this);
+  }
+
+  rtk_connectivity_watchdog_timer_.start();
+}
+
+void InertialSenseROS::stop_rtk_connectivity_watchdog_timer()
```

Comment:
I included the stop method for completion, but I don't see any need to call it.  Let me know if you don't agree.